jumanji/environments/routing/multi_cvrp/env.py

```diff
@@ -0,0 +1,508 @@
+            num_customers=20,
+            num_vehicles=2,
+        )
+
+        self._max_capacity = self._generator._max_capacity
+        self._map_max = self._generator._map_max
+        self._customer_demand_max = self._generator._customer_demand_max
+        self._max_start_window = self._generator._max_start_window
+        self._max_end_window = self._generator._max_end_window
+        self._time_window_length = self._generator._time_window_length
+        self._early_coef_rand = self._generator._early_coef_rand
+        self._late_coef_rand = self._generator._late_coef_rand
+        self._num_customers = self._generator._num_customers
+        self._num_vehicles = self._generator._num_vehicles
+
+        # Create reward function used for computing rewards
+        self._reward_fn = reward_fn or DenseReward(
+            self._num_vehicles, self._num_customers, self._map_max
+        )
+
+        # Create viewer used for rendering
+        self._viewer = viewer or MultiCVRPViewer(
+            name="MultiCVRP",
+            num_vehicles=self._num_vehicles,
+            num_customers=self._num_customers,
+            map_max=self._map_max,
+            render_mode="human",
+        )
+
+        # From the paper: All distances are represented by Euclidean distances in the plane,
+        # and the speeds of all  vehicles are assumed to be identical (i.e., it takes one
+        # unit of time to travel one unit of distance)
+        self._speed: int = 1
+
+        self._max_local_time = (
+            max_single_vehicle_distance(self._map_max, self._num_customers)
+            / self._speed
+        )
+
+    def __repr__(self) -> str:
+        return f"MultiCVRP(num_customers={self._num_customers}, num_vehicles={self._num_vehicles})"
+
+    def reset(self, key: chex.PRNGKey) -> Tuple[State, TimeStep[Observation]]:
+        """
+        Resets the environment.
+
+        Args:
+            key: used to randomly generate the problem and the start node.
+
+        Returns:
+             state: State object corresponding to the new state of the environment.
+             timestep: TimeStep object corresponding to the first timestep returned by the
+                environment.
+        """
+
+        state = self._generator(key)
+
+        timestep = restart(observation=self._state_to_observation(state))
+
+        return state, timestep
+
+    def step(
+        self, state: State, actions: chex.Array
+    ) -> Tuple[State, TimeStep[Observation]]:
+        """
+        Run one timestep of the environment's dynamics.
+
+        Args:
+            state: State object containing the dynamics of the environment.
+            actions: Array containing the index of the next nodes to visit.
+
+        Returns:
+            state, timestep: Tuple[State, TimeStep] containing the next state of the environment,
+                as well as the timestep to be observed.
+        """
+        new_state = self._update_state(state, actions)
```

Comment:
In jumanji an action (even if multi-dimensional) is singular.
```suggestion
        self, state: State, action: chex.Array
    ) -> Tuple[State, TimeStep[Observation]]:
        """
        Run one timestep of the environment's dynamics.

        Args:
            state: State object containing the dynamics of the environment.
            action: Array containing the index of the next nodes to visit.

        Returns:
            state, timestep: Tuple[State, TimeStep] containing the next state of the environment,
                as well as the timestep to be observed.
        """
        new_state = self._update_state(state, action)
```